### PR TITLE
NAS-104840 / 12.0 / WebDAVFSAttachmentDelegate

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/webdav.py [hunk 1]
```diff
@@ -271,5 +275,36 @@ class Config:
...
+        return results
+
+    async def get_attachment_name(self, attachment):
+        return attachment['name']
+
+    async def delete(self, attachments):
+        for attachment in attachments:
+            await self.middleware.call('datastore.delete', 'sharing.webdav_share', attachment['id'])
+
+        await self._service_change('webdav', 'reload')
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/webdav.py]: Wouldnt we have to stop in this case?

File: src/middlewared/middlewared/plugins/webdav.py [hunk 2]
```diff
@@ -271,5 +275,36 @@ class Config:
...
+
+    async def toggle(self, attachments, enabled):
+        for attachment in attachments:
+            await self.middleware.call('datastore.update', 'sharing.webdav_share', attachment['id'],
+                                       {'webdav_enabled': enabled})
+
+        await self._service_change('webdav', 'reload')
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/webdav.py]: Shouldnt we start or stop depending on enabled?

File: src/middlewared/middlewared/plugins/webdav.py (hunk 1 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/webdav.py]: We don't stop services when exporting pools because they could be serving other pools.

Comment (william-gr) [reply src/middlewared/middlewared/plugins/webdav.py]: Ah, of course, thanks for clarifying

**Pull request merged**